src/Microsoft.AspNetCore.Razor.Language/Legacy/RazorParser.cs

```diff
@@ -37,7 +37,10 @@ public virtual RazorSyntaxTree Parse(RazorSourceDocument source)
 
             var reader = new SeekableTextReader(chars, source.FileName);
 
-            var context = new ParserContext(reader, Options.DesignTimeMode);
+            var context = new ParserContext(reader, Options.DesignTimeMode)
+            {
+                StopParsingAfterFirstDirective = Options.StopParsingAfterFirstDirective
+            };
```

Comment:
`parse single directive method to RazorSyntaxTree correct` - this is exactly what I'm suggesting.